### ISSUE TITLE
Scoped @Provide methods for Subcomponent fail to compile

### ISSUE THREAD

**Issue by feichngr** [google/dagger#163]
A `Subcomponent` using a `Module` with scoped `@Provide` methods (same scope as the root `Component`) fails to compile with an exception.

e.g.

Component:
```java
@Singleton
@Component
public interface RootComponent {
    ChildComponent childComponent();
}
```

Subcomponent:
```java
// @Singleton here doesn't help :/
@Subcomponent(modules = ChildModule.class)
public interface ChildComponent {
    Object moduleProvidedObject();
}
```

Module:
```java
@Module
public class ChildModule {
    @Singleton
    @Provides
    Object provideObject() {
        return new Object() {
            @Override
            public String toString() {
                return "this should work, right?";
            }
        };
    }
}
```

fails with this stacktrace (shortened) when build with Dagger 2.0:
```
Caused by: java.lang.IllegalArgumentException: no bindings
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
	at dagger.internal.codegen.ContributionBinding.bindingTypeFor(ContributionBinding.java:101)
	at dagger.internal.codegen.ComponentGenerator.writeField(ComponentGenerator.java:528)
	at dagger.internal.codegen.ComponentGenerator.writeFields(ComponentGenerator.java:506)
	at dagger.internal.codegen.ComponentGenerator.writeComponent(ComponentGenerator.java:322)
	at dagger.internal.codegen.ComponentGenerator.write(ComponentGenerator.java:232)
	at dagger.internal.codegen.ComponentGenerator.write(ComponentGenerator.java:122)
	at dagger.internal.codegen.SourceFileGenerator.generate(SourceFileGenerator.java:53)
	at dagger.internal.codegen.ComponentProcessingStep.process(ComponentProcessingStep.java:82)
	at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:228)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
	at com.sun.tools.javac.main.Main.compile(Main.java:523)
	... 73 more
```

**Comment by gk5885** [comment]
This _shouldn't_ work because `ChildComponent` is an unscoped subcomponent, but `provideObject` declares a scoped binding.  (`@Singleton` bindings can only be present in `@Singleton` (sub)components).  That said, the stack trace was an error, which has been addressed for 2.0.1.  You'll get a proper error message now.

**Comment by ypresto** [comment]
> That said, the stack trace was an error, which has been addressed for 2.0.1. You'll get a proper error message now.

This is not fixed in neither 2.0.1 nor 2.0.2.

**Comment by ronshapiro** [comment]
Can you check if it's fixed in the `2.1-SNAPSHOT`?
